opentreemap/treemap/udf.py

```diff
@@ -990,132 +1202,154 @@ def __getitem__(self, key):
         if udf.iscollection:
             return self.collection_fields.get(key, [])
         else:
-            if super(UDFDictionary, self).__contains__(key):
-                v = super(UDFDictionary, self).__getitem__(key)
-                try:
-                    return udf.clean_value(v)
-                except:
-                    return v
-            else:
+            try:
+                value = super(UDFDictionary, self).__getitem__(key)
+                return udf.clean_value(value)
+            except KeyError:
                 return None
 
-    def __setitem__(self, key, val):
+    def __setitem__(self, key, val, dtd=None):
         udf = self._get_udf_or_error(key)
 
         if udf.iscollection:
-            self.instance.dirty_collection_udfs[key] = True
-            # HStoreDict cleans values in-place, so we need to do a deep-copy
+            self.instance.dirty_collection_udfs.add(key)
+            # Don't let the caller update collection-subfields on the fly.
```

Comment:
The comment was sort of notes to myself to try to explain the comment that used to be there. Checking into the behavior.